compiler/qsc_eval/src/backend.rs

```diff
@@ -13,32 +13,84 @@ use crate::val::Value;
-    fn reset(&mut self, q: usize);
-    fn rx(&mut self, theta: f64, q: usize);
-    fn rxx(&mut self, theta: f64, q0: usize, q1: usize);
-    fn ry(&mut self, theta: f64, q: usize);
-    fn ryy(&mut self, theta: f64, q0: usize, q1: usize);
-    fn rz(&mut self, theta: f64, q: usize);
-    fn rzz(&mut self, theta: f64, q0: usize, q1: usize);
-    fn sadj(&mut self, q: usize);
-    fn s(&mut self, q: usize);
-    fn swap(&mut self, q0: usize, q1: usize);
-    fn tadj(&mut self, q: usize);
-    fn t(&mut self, q: usize);
-    fn x(&mut self, q: usize);
-    fn y(&mut self, q: usize);
-    fn z(&mut self, q: usize);
-    fn qubit_allocate(&mut self) -> usize;
-    fn qubit_release(&mut self, q: usize);
-    fn capture_quantum_state(&mut self) -> (Vec<(BigUint, Complex<f64>)>, usize);
-    fn qubit_is_zero(&mut self, q: usize) -> bool;
+    fn ccx(&mut self, _ctl0: usize, _ctl1: usize, _q: usize) {
+        unimplemented!();
+    }
+    fn cx(&mut self, _ctl: usize, _q: usize) {
+        unimplemented!();
+    }
+    fn cy(&mut self, _ctl: usize, _q: usize) {
+        unimplemented!();
+    }
+    fn cz(&mut self, _ctl: usize, _q: usize) {
+        unimplemented!();
+    }
+    fn h(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn m(&mut self, _q: usize) -> Self::ResultType {
+        unimplemented!();
+    }
+    fn mresetz(&mut self, _q: usize) -> Self::ResultType {
+        unimplemented!();
+    }
+    fn reset(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn rx(&mut self, _theta: f64, _q: usize) {
+        unimplemented!();
+    }
+    fn rxx(&mut self, _theta: f64, _q0: usize, _q1: usize) {
+        unimplemented!();
+    }
+    fn ry(&mut self, _theta: f64, _q: usize) {
+        unimplemented!();
+    }
+    fn ryy(&mut self, _theta: f64, _q0: usize, _q1: usize) {
+        unimplemented!();
+    }
+    fn rz(&mut self, _theta: f64, _q: usize) {
+        unimplemented!();
+    }
+    fn rzz(&mut self, _theta: f64, _q0: usize, _q1: usize) {
+        unimplemented!();
+    }
+    fn sadj(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn s(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn swap(&mut self, _q0: usize, _q1: usize) {
+        unimplemented!();
+    }
+    fn tadj(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn t(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn x(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn y(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn z(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn qubit_allocate(&mut self) -> usize {
+        unimplemented!();
+    }
+    fn qubit_release(&mut self, _q: usize) {
+        unimplemented!();
+    }
+    fn capture_quantum_state(&mut self) -> (Vec<(BigUint, Complex<f64>)>, usize) {
+        unimplemented!();
+    }
+    fn qubit_is_zero(&mut self, _q: usize) -> bool {
+        unimplemented!();
```

Comment:
this may need to be overriden by you later to return `true` otherwise qubit release in the evaluator will fail, but you will also probably want qubit allocate and release too, so they all go together.